pkg/option/option.go

```diff
@@ -115,6 +115,9 @@ func (om OptionMap) DeepCopy() OptionMap {
 	return cpy
 }
 
+// BoolOptions member functions with external access do not require
+// locking by the caller, while functions with internal access presume
+// the caller to have taken care of any locking needed.
```

Comment:
I've seen a style before in golang that names functions `fooLocked()`, `fooRLocked()` to indicate that they should be called while holding the lock. Might be a bit weighty in this case, but something to consider.

Reply:
I see this was already discussed above ;) Feel free to disregard.